packages/vite/src/node/plugins/resolve.ts

```diff
@@ -1025,22 +1011,29 @@ export function resolvePackageEntry(
 
     for (let entry of entryPoints) {
       // make sure we don't get scripts when looking for sass
+      let skipPackageJson = false
       if (
         options.mainFields[0] === 'sass' &&
         !options.extensions.includes(path.extname(entry))
       ) {
         entry = ''
-        options.skipPackageJson = true
```

Comment:
This was modifying the `options` object for every `entry`. I think there was a bug here we never catched. I refactored `skipPackageJson` to pass it directly to `tryFsResolve` now.